packages/dds/map/src/test/mocha/directory.spec.ts

```diff
@@ -29,17 +33,16 @@ function createConnectedDirectory(id: string, runtimeFactory: MockContainerRunti
     return directory;
 }
 
-function createLocalMap(id: string) {
+function createLocalMap(id: string): ISharedMap {
     const factory = new MapFactory();
     return factory.create(new MockFluidDataStoreRuntime(), id);
 }
 
-// eslint-disable-next-line @typescript-eslint/ban-types
-async function populate(directory: SharedDirectory, content: object) {
+async function populate(directory: SharedDirectory, content: unknown): Promise<void> {
     const storage = new MockSharedObjectServices({
         header: JSON.stringify(content),
     });
-    return directory.load(storage);
+    await directory.load(storage);
```

Comment:
Is linting forcing this change? I haven't explored the scenario in detail in JS but in C# the equivalent change would be functional, the promise would now "pause" inside this function if it did anything asynchronous, instead of letting the await of the call to `populate()` (or wherever it happened up in the call stack) be the one to "pause". In C# it also introduces overhead to instantiate the magic-state-machine-behind-the-scenes that enables async functionality... here I'd imagine it could have an effect on the JS turn system.

Reply:
No, this was unrelated, and probably shouldn't have been pushed. I'll revert this.

That's a good point about the JS turn impact. I suspect it's negligible in the single await case, but it's good to keep in mind. Stylistically I like awaiting over returning a promise simply because it makes the async nature of the code more immediately apparent to the reader. But either way, it's unrelated to these changes, so I'll revert 🙂

Reply:
```suggestion
    return directory.load(storage);
```

Reply:
When I was coding in C# the heuristic I ended up adopting is: if the function does nothing after awaiting, you probably want to just return the Promise object and let someone else await when appropriate; `await` otherwise, because you don't have a choice 😄.